invenio_vocabularies/records/systemfields/relations.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 CERN.
+#
+# Invenio-Records-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Relations system fields."""
+
+from flask import current_app
+from invenio_records.systemfields import RelationsField
+from werkzeug.local import LocalProxy
+
+from invenio_vocabularies.services.custom_fields import VocabularyCF
+
+from ..api import Vocabulary
+
+
+class CustomFieldsRelation(RelationsField):
+    """Relation field to manage custom fields.
+
+    Iterates through all configured custom fields and collects the ones
+    defining a relation dependency e.g vocabularies.
+    """
+
+    def __init__(self, fields_var):
+        """Initialize the field."""
+        super().__init__()
+        self._fields_var = fields_var
+        self._fields = LocalProxy(lambda: self._load_custom_fields_relations())
+
+    def _load_custom_fields_relations(self):
+        custom_fields = current_app.config.get(self._fields_var, {})
+
+        relations = {}
+        for cf in custom_fields:
+            if getattr(cf, "relation_cls", None):
+                relations[cf.name] = cf.relation_cls(
+                    f"custom_fields.{cf.name}",
```

Comment:
will this be a problem with object fields?
e.g. 
```json
"cern:experiment": {
    "cern:experiment_name": VocabularyField...
}
```

Reply:
Yes, we can tackle this as separate issue.